src/main/java/com/adobe/target/edge/client/ondevice/DefaultRuleLoader.java

```diff
@@ -225,7 +233,13 @@ protected boolean loadRules(ClientConfig clientConfig) {
     try {
       TargetExceptionHandler handler = clientConfig.getExceptionHandler();
       GetRequest request = generateRequest(clientConfig);
+      TimingTool timer = new TimingTool();
+      timer.timeStart(TIMING_EXECUTE_REQUEST);
       HttpResponse<OnDeviceDecisioningRuleSet> response = executeRequest(request);
+      long artifactDownloadTime =
+          ((long) MathUtils.roundDouble(timer.timeEnd(TIMING_EXECUTE_REQUEST), DECIMAL_PLACES));
+      artifactDownloadTimeStore.add(artifactDownloadTime);
```

Comment:
Instead of storing this metric inside a new data structure, can we just overload the `TelemetryService.addTelemetry` method and call it directly here?
For reference, this is the entry that we store for this use case in the Node.js SDK:
```
{
  "requestId": "ArtifactDownload",
  "timestamp" [current epoch timestamp in milliseconds],
  "execution": [double type download time]
}
```

*Note that we use a static requestId for this entry type

Reply:
In case you want to look at the Node.js impl:  https://github.com/adobe/target-nodejs-sdk/blob/7437aa12c4b4b9ddb1caa6f3dbb33bae01f1ac5f/packages/target-decisioning-engine/src/artifactProvider.js#L167

Reply:
When we first load the app artifactFetch time gets set in the memory store, in the subsequent request addTelemetry() method gets called for ODD where telemetry entry is set. 
This is how it behaved when I tested the flow in the sample app.

Reply:
I have kept a separate telemetry entry for the artifact download time with timestamp & hardcoded requestId.
Since Telemetry Service is Singleton in nature, I needed to have a thread safe global variable to store the artifact download time.